streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```diff
@@ -49,12 +49,17 @@
 
     private final Producer<byte[], byte[]> producer;
     private final Map<TopicPartition, Long> offsets;
+    private String streamTaskId = null;
 
     public RecordCollector(Producer<byte[], byte[]> producer) {
         this.producer = producer;
         this.offsets = new HashMap<>();
     }
 
+    void setStreamTask(String streamTaskId) {
```

Comment:
Could we just pass in the taskId as part of the constructor? And for unit tests we can simply set a dummy taskId here.